From e778093683a32d4e3e139f664aa289e55faac7db Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Fri, 11 Apr 2025 17:51:29 -0700
Subject: [PATCH 7/9] Change refs to GTEST_BOTH_LIBRARIES to explicit
 libraries.

* This variable is old and only comes from the GTest find module, not by the upstream module itself.
* Modern cmake norms are just to depend on the library aliases themselves.
---
 clients/benchmarks/CMakeLists.txt | 4 ++--
 clients/common/CMakeLists.txt     | 5 ++---
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/clients/benchmarks/CMakeLists.txt b/clients/benchmarks/CMakeLists.txt
index b46879a2..dddcbf58 100644
--- a/clients/benchmarks/CMakeLists.txt
+++ b/clients/benchmarks/CMakeLists.txt
@@ -57,14 +57,14 @@ target_link_libraries( rocblas-bench PRIVATE rocblas_clients_testing_common rocb
 if( BUILD_FORTRAN_CLIENTS )
   target_link_libraries( rocblas-bench PRIVATE rocblas_fortran_client )
 endif( )
-target_link_libraries( rocblas-bench PRIVATE roc::rocblas hip::host hip::device ${BLAS_LIBRARY} ${GTEST_BOTH_LIBRARIES} )
+target_link_libraries( rocblas-bench PRIVATE roc::rocblas hip::host hip::device ${BLAS_LIBRARY} GTest::gtest GTest::gtest_main )
 
 if( BUILD_WITH_TENSILE )
   target_link_libraries( rocblas-gemm-tune PRIVATE rocblas_clients_testing_common rocblas_clients_common )
   if( BUILD_FORTRAN_CLIENTS )
     target_link_libraries( rocblas-gemm-tune PRIVATE rocblas_fortran_client )
   endif( )
-  target_link_libraries( rocblas-gemm-tune PRIVATE roc::rocblas hip::host hip::device ${BLAS_LIBRARY} ${GTEST_BOTH_LIBRARIES} )
+  target_link_libraries( rocblas-gemm-tune PRIVATE roc::rocblas hip::host hip::device ${BLAS_LIBRARY} GTest::gtest GTest::gtest_main )
 endif()
 
 if (NOT WIN32)
diff --git a/clients/common/CMakeLists.txt b/clients/common/CMakeLists.txt
index 7ab67d91..f49231b4 100644
--- a/clients/common/CMakeLists.txt
+++ b/clients/common/CMakeLists.txt
@@ -47,7 +47,6 @@ function( rocblas_client_library_settings lib_target_ )
     $<BUILD_INTERFACE:${HIP_INCLUDE_DIRS}>
     $<BUILD_INTERFACE:${BLAS_INCLUDE_DIR}>
     $<BUILD_INTERFACE:${BLIS_INCLUDE_DIR}> # may be blank if not used
-    $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
   )
 
   target_include_directories( ${lib_target_}
@@ -162,7 +161,7 @@ target_compile_definitions( rocblas_clients_common PRIVATE ROCBLAS_REFERENCE_LIB
 
 
 # target is not linked as OBJECT library but we want any interface flags from dependencies
-target_link_libraries( rocblas_clients_common roc::rocblas hip::host hip::device ${GTEST_BOTH_LIBRARIES} ${COMMON_LINK_LIBS} )
-target_link_libraries( rocblas_clients_testing_common roc::rocblas hip::host hip::device ${BLAS_LIBRARY} ${GTEST_BOTH_LIBRARIES} ${COMMON_LINK_LIBS} )
+target_link_libraries( rocblas_clients_common roc::rocblas hip::host hip::device GTest::gtest GTest::gtest_main ${COMMON_LINK_LIBS} )
+target_link_libraries( rocblas_clients_testing_common roc::rocblas hip::host hip::device ${BLAS_LIBRARY} GTest::gtest GTest::gtest_main ${COMMON_LINK_LIBS} )
 
 
-- 
2.45.1.windows.1

